tsdb/index/tsi1/index.go

```diff
@@ -106,10 +114,26 @@ func NewIndex() *Index {
 		MaxLogFileSize:    DefaultMaxLogFileSize,
 		CompactionEnabled: true,
 
-		logger: zap.New(zap.NullEncoder()),
+		logger:  zap.New(zap.NullEncoder()),
+		version: Version,
 	}
 }
 
+// An IncompatibleVersionError is returned when attempting to read from an
+// incompatible tsi1 manifest file.
+type IncompatibleVersionError error
+
+// ErrIncompatibleVersion returns an IncompatibleVersionError if the provided
+// manifest is incompatible with the current tsi1 package.
+func ErrIncompatibleVersion(m *Manifest) IncompatibleVersionError {
+	// If we don't have an explicit version in the manifest file then we know
+	// it's not compatible with the latest tsi1 Index.
+	if m.Version != Version {
+		return IncompatibleVersionError(errors.New("incompatible tsi1 index MANIFEST"))
+	}
+	return nil
+}
+
```

Comment:
I usually think of `Err` as the error type instead of the validation function. Can you do something like:

```go
// ErrIncompatibleVersion is returned when attempting to read from an
// incompatible tsi1 manifest file.
var ErrIncompatibleVersion = errors.New("incompatible tsi1 index MANIFEST")

// ErrIncompatibleVersion returns an IncompatibleVersionError if the provided
// manifest is incompatible with the current tsi1 package.
func (m *Manifest) Validate() error {
	// If we don't have an explicit version in the manifest file then we know
	// it's not compatible with the latest tsi1 Index.
	if m.Version != Version {
		return ErrIncompatibleVersion
	}
	return nil
}
```

Reply:
Sure.

Reply:
Though I should add that error _types_ should end in `...Error` and error _values_ should begin with `Err...` but I know what you mean, and in this case an explicit type was overkill and I never checked it anywhere.